src/components/Admin/Users/index.js

```diff
@@ -15,29 +18,100 @@ import useFetch from '../../../hooks/useFetch';
 const { USERS_ALL } = routes;
 
 export default function Users() {
-  // fetch users
-  const [{ data: msg, isLoading }] = useFetch('/.netlify/functions/getUsers');
+  const [modalVisible, setModalVisible] = useState(false);
+  const [deletingUser, setDeletingUser] = useState(false);
+  const [userToDelete, setUserToDelete] = useState(null);
+  const [userData, setUserData] = useState(null);
+  const [loading, setLoading] = useState(false);
+
+  const toggleModal = () => setModalVisible(!modalVisible);
+
+  const deleteUser = userId => {
+    setUserToDelete(userId);
+    toggleModal();
+  };
+
+  const confirmDelete = async () => {
+    try {
+      setDeletingUser(true);
+      await axios.delete('/.netlify/functions/deleteUser', {
+        data: {
+          userId: userToDelete,
+        },
+      });
+      const { data: newUsers } = await axios.get(
+        '/.netlify/functions/getUsers'
+      );
+      setUserData(newUsers);
+      setDeletingUser(false);
+      setUserToDelete(null);
+      toggleModal();
+      return message.success('User and all related data successfully deleted');
+    } catch (err) {
+      setDeletingUser(false);
+      setUserToDelete(null);
+      toggleModal();
+      return message.error(`Sorry, there was an error: ${err}`);
+    }
+  };
+
+  // // fetch users
+  // const data = useFetch('/.netlify/functions/getUsers');
+
+  // useEffect(() => {
+  //   async function fetchData() {
+  //     // You can await here
+  //     const response = await MyAPI.getData(someId);
+  //     // ...
+  //   }
+  //   fetchData();
+
+  const [{ data: msg, isLoading }] = useFetch(
+    '/.netlify/functions/getUsers',
+    userData
+  );
+
+  useEffect(() => {
+    if (msg) {
+      setLoading(isLoading);
+      setUserData(msg);
+    }
+  }, [isLoading, msg]);
 
   // create table friendly data sets
   const users =
-    msg &&
-    msg.msg.map(({ _id, name, email, role, status }) => ({
+    userData &&
+    userData.msg.map(({ _id, name, email, role, status }) => ({
       key: _id,
       name,
       email,
       level: role,
       status,
+      actions: { id: _id, status, deleteUser },
     }));
 
-  console.log(users);
   return (
     <Switch>
-      {isLoading && <Loading />}
+      {loading && <Loading />}
       <Route
         exact
         path={USERS_ALL}
         render={props => (
-          <Table columns={usersCol} dataSource={users} {...props} />
+          <>
+            <Table columns={usersCol} dataSource={users} {...props} />
```

Comment:
antD table has loading attribute that shows a nice loading wheel. for now the table shows No Data icon until the loading finish which can be misleading